### PR TITLE
Feature/11887 header

### PR DESCRIPTION
**Pull request by erochOnGit**

### CODE REVIEW

File: src/web/theme/modules/Header/Header.scss
Comment (real34) [inline src/web/theme/modules/Header/Header.scss]: It should be a class. Please prevent targeting direct HTML elements.

**Review by real34 — CHANGES_REQUESTED** [review]
Could you pleas provide further information on what you are trying to achieve?

From what I see, the « navigation steps » seems to be a complex implementation for something that could have been standard `<Link to="/xxxx">` and leveraging React Router’s features.

File: src/web/theme/modules/Header/Navigation/Menu/Menu.scss
Comment (real34) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.scss]: This selector is too specific. Could you think of another way to achieve styling using simpler rules?

File: src/web/theme/modules/Header/Navigation/Menu/Menu.js
Comment (real34) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.js]: Please follow the BEM conventions here, and use a variant: `menu menu--open`

File: src/web/theme/modules/Header/Navigation/Menu/Menu.js
Comment (real34) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.js]: Same here about BEM usage.

Could we improve the documentation with BEM instructions to help?

File: src/web/theme/modules/Header/Navigation/Menu/Menu.scss
Comment (aloross) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.scss]: `padding: 1em;` is enough.

File: src/web/theme/modules/Header/Navigation/Menu/Menu.scss
Comment (aloross) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.scss]: `padding: 1em;` is enough.

File: src/web/theme/modules/Header/Navigation/Menu/Menu.scss
Comment (aloross) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.scss]: `menu__button` should be enough here. If not, there might be an issue with another css rule.

File: src/web/theme/modules/Header/Navigation/Menu/Menu.scss
Comment (aloross) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.scss]: Are those `top` and `left` properties doing anything ?

**Review by aloross — CHANGES_REQUESTED** [review]
I don't understand the need for navigation steps. Why Can't you just use react-router [`NavLink`](https://reacttraining.com/react-router/web/api/NavLink) with `activeClassName` ?

File: src/web/theme/modules/Header/Navigation/Menu/Menu.js
Comment (milkae) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.js]: If it's a question of classname, you should put the condition in the classname instead of duplicated the entire component.

File: src/web/theme/modules/Header/Navigation/Navigation.js
Comment (milkae) [inline src/web/theme/modules/Header/Navigation/Navigation.js]: You don't have to use a function as initial state

File: src/web/theme/modules/Header/Navigation/Navigation.js
Comment (milkae) [inline src/web/theme/modules/Header/Navigation/Navigation.js]: Coul you extracts that part of logic making it more readable?

**Review by milkae — CHANGES_REQUESTED** [review]

File: src/web/theme/modules/Header/Header.scss
Comment (hihuz) [inline src/web/theme/modules/Header/Header.scss]: Nit : we usually don't specifiy unit when value is 0.

File: src/web/theme/modules/Header/Navigation/Menu/Menu.js
Comment (hihuz) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.js]: You don't need the brackets if you pass a string as prop, e.g. `icon="menu"`

File: src/web/theme/modules/Header/Navigation/Menu/Menu.js
Comment (hihuz) [reply src/web/theme/modules/Header/Navigation/Menu/Menu.js]: definitely

File: src/web/theme/modules/Header/Navigation/Menu/Menu.js
Comment (hihuz) [inline src/web/theme/modules/Header/Navigation/Menu/Menu.js]: camel case should be avoided in class names I think, you could use `menu__nav-button` here

**Review by hihuz — CHANGES_REQUESTED** [review]
My comments are minor but some others before are more important, I'll approve once the changes are made

**Pull request merged**
